shared-workers.md

```diff
@@ -27,11 +27,28 @@ There has been increasing [developer](https://github.com/GoogleChromeLabs/privac
 
 ## [Maintaining a Session](https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/issues/124)
 
-A developer embeds chat.com on two of their sites site-a.com and site-b.com. chat.com uses Shared Workers to maintain a user session.
+Some websites allow users to chat with the site's customer service.
+For example, the US internet provider Comcast splits parts of its services across business.comcast.com, xfinity.com, and more.
+In some cases, pieces of a user's natural journey while interacting with the customer service agent lead them to different but associated sites.
+This could be implemented by putting a session ID into a cookie, and that would continue to work while cookies are partitioned if the user grants storage access.
+However, it's more natural and common for sites to use a SharedWorker to manage the chat session.
+This proposal allows that to work after the same use permission grant.
+
+A website, company-chat.example, offers a way to maintain active chat session between any sites the embed it using SharedWorkers.
+Before storage partitioning this was possible, but after storage partitioning when company-chat.example is embedded on different sites (company-site1.example or company-site2.example) and instantiates a SharedWorker it would no longer be shared.
+By prompting the user for permission via `document.requestStorageAccess({SharedWorker: true})` and then instantiating the SharedWorker via the returned handle the worker can be shared across partitioned third-party contexts.
 
 ## [Transfering an ArrayBuffer](https://groups.google.com/a/chromium.org/g/blink-dev/c/inRN8tI49O0/m/Q_TE0cw4AAAJ)
 
-A developer embeds worker.com on two of their sites site-a.com and site-b.com. worker.com uses Shared Workers to transfer ArrayBuffers via postMessage.
+Similar to the above case, but perhaps the customer service chat has a video option that can start embedded in the page but could also be popped out into its own window.
+This could be implemented by putting a session ID into the query params of the opened window.
+However, to allow a more natural transition without audio/video glitches it would be easier to have a SharedWorker seamlessly transition the call over.
+This proposal allows that to work after a storage access permission grant.
+
+A website, company-videochat.example, wants to track and resume a video call in a first party context using SharedWorkers. `window.postMessage(...)` won't work as it would require inefficient cloning of significant data.
```

Comment:
I think I'm not familiar enough with the way transferring ArrayBuffers works: do they transfer between a site and its own `SharedWorker` but not across cross-site iframe boundaries? I guess that makes sense, since different sites are likely to be running in different processes. It'd be good to say that explicitly so nobody else stumbles over it.

Reply:
window.postMessage essentially forces things to be serialized/deserialized as JSON, where SharedWorker can take complex objects like ArrayBuffers intact.

Reply:
I had no idea that was possible, cool! The documentation makes it sound like it works to transfer between threads and not processes, yeah.

Reply:
Okay, so it looks like you can send them cross origin. I suppose the issue is that this does require cloning the data if you need it in both renderers at once.